sdk/contentsafety/ai-content-safety-rest/test/public/contentSafety.spec.ts

```diff
@@ -0,0 +1,179 @@
+
+describe("Content Safety Client Test", () => {
+  let recorder: Recorder;
+  let client: ContentSafetyClient;
+  function sleep(time: number): Promise<NodeJS.Timer> {
+    return new Promise((resolve) => setTimeout(resolve, time));
+  }
+
+  beforeEach(async function (this: Context) {
+    recorder = await createRecorder(this);
+    client = await createClient(recorder);
+  });
+
+  afterEach(async function () {
+    await recorder.stop();
+  });
+
+  it("analyze text", async function () {
+    const response = await client.path("/text:analyze").post({
+      body: {
+        text: "You are an idiot",
+        categories: ["Hate"],
+      },
+    });
+    if (isUnexpected(response)) {
+      throw new Error(response.body?.error.message);
+    }
+    assert.strictEqual(response.status, "200");
+    assert.equal(response.body.hateResult?.category, "Hate");
+    assert.notExists(response.body.selfHarmResult);
+  });
+
+  it("analyze image", async function () {
+    const response = await client.path("/image:analyze").post({
+      body: {
+        image: {
+          content: imageBase64Str,
+        },
+        categories: ["Sexual"],
+      },
+    });
+    if (isUnexpected(response)) {
+      throw new Error(response.body?.error.message);
+    }
+    assert.strictEqual(response.status, "200");
+    assert.equal(response.body.sexualResult?.category, "Sexual");
+    assert.notExists(response.body.violenceResult);
+  });
+
+  it("manage blocklist", async function () {
+    const blocklistName = "TestBlocklist";
+    const blockItemText1 = "k*ll";
+    const blockItemText2 = "h*te";
+
+    // create blocklist
+    const createBlockListResponse = await client
+      .path("/text/blocklists/{blocklistName}", blocklistName)
+      .patch({
+        contentType: "application/merge-patch+json",
+        body: {
+          description: "test",
+        },
+      });
+    if (isUnexpected(createBlockListResponse)) {
+      throw new Error(createBlockListResponse.body?.error.message);
+    }
+    assert.strictEqual(createBlockListResponse.status, "201");
+    assert.equal(createBlockListResponse.body.blocklistName, blocklistName);
+
+    // add block items
+    const addBlockItemsResponse = await client
+      .path("/text/blocklists/{blocklistName}:addBlockItems", blocklistName)
+      .post({
+        body: {
+          blockItems: [
+            {
+              description: "Test block item 1",
+              text: blockItemText1,
+            },
+            {
+              description: "Test block item 2",
+              text: blockItemText2,
+            },
+          ],
+        },
+      });
+    if (isUnexpected(addBlockItemsResponse)) {
+      throw new Error(addBlockItemsResponse.body?.error.message);
+    }
+    assert.strictEqual(addBlockItemsResponse.status, "200");
+    assert.isArray(addBlockItemsResponse.body.value);
+
+    // sleep 30s to wait for block items to be ready
+    await sleep(30000);
```

Comment:
It will be successfully added when the API returns 200. But there needs 5 minutes to wait before it can be used in Analyze Text API. I used `isRecordMode` to skip sleep in playback mode.